bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusGlobalHandler.java

```diff
@@ -0,0 +1,105 @@
+import org.openhab.binding.qbus.internal.QbusBridgeHandler;
+import org.openhab.binding.qbus.internal.protocol.QbusCommunication;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+
+/**
+ * The {@link QbusGlobalHandler} is used in other handlers, to share the functions.
+ *
+ * @author Koen Schockaert - Initial Contribution
+ */
+
+@NonNullByDefault
+public abstract class QbusGlobalHandler extends BaseThingHandler {
+
+    public QbusGlobalHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Get Bridge communication
+     *
+     * @param type
+     * @param globalId
+     * @return
+     */
+    public @Nullable QbusCommunication getCommunication(String type, @Nullable Integer globalId) {
+        QbusBridgeHandler qBridgeHandler = null;
+        if (globalId != null) {
+            qBridgeHandler = getBridgeHandler(type, globalId);
+        }
+
+        if (qBridgeHandler == null) {
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                    "No bridge handler initialized for " + type + " with id " + globalId + ".");
+            return null;
+        }
+        QbusCommunication qComm = qBridgeHandler.getCommunication();
+        return qComm;
+    }
+
+    /**
+     * Get the Bridge handler
+     *
+     * @param type
+     * @param globalId
+     * @return
+     */
+    public @Nullable QbusBridgeHandler getBridgeHandler(String type, @Nullable Integer globalId) {
+        Bridge qBridge = getBridge();
+        if (qBridge == null) {
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                    "No bridge initialized for " + type + " with ID " + globalId);
+            return null;
+        }
+        QbusBridgeHandler qBridgeHandler = (QbusBridgeHandler) qBridge.getHandler();
+        return qBridgeHandler;
+    }
+
+    /**
+     *
+     * @param qComm
+     * @param type
+     * @param globalId
+     */
+    public void restartCommunication(QbusCommunication qComm, String type, @Nullable Integer globalId) {
+        try {
+            qComm.restartCommunication();
+        } catch (InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.toString());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.toString());
+        }
+
+        if (!qComm.communicationActive()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Communication socket error");
+            return;
+        }
+
+        QbusBridgeHandler qBridgeHandler = getBridgeHandler(type, globalId);
+        if (qBridgeHandler != null) {
+            qBridgeHandler.bridgeOnline();
+        }
```

Comment:
You might want to call `qBridgeHandler.bridgeOnline()` only if no exception occurred. Otherwise, the Thing will be set to OFFLINE and then, `qBridgeHandler.bridgeOnline()` will be called.